src/NumericalAlgorithms/FiniteDifference/HighOrderFluxCorrection.hpp

```diff
@@ -94,10 +95,24 @@ void cartesian_high_order_fluxes_using_nodes(
         Dim, Variables<tmpl::list<::Tags::Flux<
                  EvolvedVarsTags, tmpl::size_t<Dim>, Frame::Inertial>...>>>&
         ghost_cell_inertial_flux,
-    const Mesh<Dim>& subcell_mesh, const size_t number_of_ghost_cells) {
-  static constexpr size_t max_correction_order = 10;
-  static constexpr size_t correction_width = CorrectionOrder / 2 - 1;
-  static_assert(CorrectionOrder <= max_correction_order);
+    const Mesh<Dim>& subcell_mesh, const size_t number_of_ghost_cells,
+    [[maybe_unused]] const std::optional<
+        std::array<gsl::span<std::uint8_t>, Dim>>& reconstruction_order =
+        std::nullopt) {
+  constexpr int max_correction_order = 10;
+  static_assert(static_cast<int>(DerOrder) <= max_correction_order);
+  static_assert(DerOrder != DerivativeOrder::OneHigherThanRecons and
+                DerOrder != DerivativeOrder::OneHigherThanReconsButFiveToFour);
+  constexpr size_t stencil_size =
```

Comment:
you've static asserted that the two cases where the cast is less than zero cannot occur so simplify?

Reply:
okay, I see you've removed the static_assert in the 3rd commit; so maybe don't add it in the first commit....